src/Modules/Manage/Dnn.PersonaBar.Users/Components/UsersController.cs

```diff
@@ -534,6 +541,15 @@ private static bool IsAdmin(UserInfo user, PortalSettings portalSettings)
                     includeSuperUsers);
         }
 
+        private PortalSettings GetPortalSettings(int portalId)
+        {
+            var portalSettings = new PortalSettings(portalId);
+            var portalAliases = PortalAliasController.Instance.GetPortalAliasesByPortalId(portalId);
```

Comment:
You are using `PortalAliasController.Instance`. Please turn it into instance field. Isn't is also necessary to put in constructor to mock in tests?

Reply:
Currently unit test is added for Add-Roles, I think it should be done in separate task